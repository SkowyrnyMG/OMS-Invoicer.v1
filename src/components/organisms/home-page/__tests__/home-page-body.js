import React from 'react';

import {
  renderWithReduxRouter,
  snapShotTest,
  screen,
  userEvent,
} from 'utils/tests/test-helper';
import HomePageBody from '../body/home-page-body';

describe('HomePageBody', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<HomePageBody />);
  });

  test('should should redirect to the register page after if clicked on register link', () => {
    const { history } = renderWithReduxRouter(<HomePageBody />);

    const initialPath = history.location.pathname;
    const registerLinkNode = screen.getByRole('link', { name: /register/i });

    userEvent.click(registerLinkNode);

    const finalPath = history.location.pathname;

    expect(initialPath).not.toBe(finalPath);
    expect(finalPath).toBe('/register');
  });
});
