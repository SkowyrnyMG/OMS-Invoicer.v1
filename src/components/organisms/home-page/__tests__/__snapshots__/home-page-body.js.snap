// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageBody should display in the document with correct styles 1`] = `
.c19 {
  display: inline-block;
  padding: 1rem 2rem;
  margin-top: 0;
  font-weight: 700;
  color: #000 !important;
  border-left: 1px solid #F0F0FD;
  background: none;
  -webkit-transition: -webkit-transform 0.25s;
  -webkit-transition: transform 0.25s;
  transition: transform 0.25s;
  border: none;
}

.c19:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 20px;
  font-weight: 700;
}

.c21 {
  width: 6rem;
}

.c1 {
  grid-column: -1 / 1;
  display: grid;
  grid-template-columns: 5rem [start] 1fr [end] 5rem;
  grid-column-gap: 3%;
  padding: 20rem 0;
  position: relative;
  background-image: url(bg-pattern.png);
  background-size: cover;
  background-position: center;
}

.c1 > * {
  grid-column: start / end;
}

.c1 {
  padding: 10rem 0;
}

.c8 {
  padding: 5rem 0;
  display: grid;
  grid-template-columns: 5rem [start] repeat(2,1fr) [end] 5rem;
  grid-column: 1 / -1 !important;
  grid-gap: 3%;
  grid-row-gap: 5%;
  min-height: 60rem;
  background-color: #446DF644;
  margin: 0;
  padding-top: 10rem;
}

.c8 > * {
  grid-column: start / end;
}

.c14 {
  padding: 5rem 0;
  display: grid;
  grid-template-columns: 5rem [start] repeat(2,1fr) [end] 5rem;
  grid-column: 1 / -1 !important;
  grid-gap: 3%;
  grid-row-gap: 5%;
  min-height: 60rem;
  background-color: #F8F8FF;
  margin: 0;
  padding-top: 10rem;
}

.c14 > * {
  grid-column: start / end;
}

.c9 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  grid-column-gap: 10rem;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  position: relative;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 34px;
}

.c15 {
  position: relative;
  margin: 10rem 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 34px;
}

.c12 {
  position: absolute;
  line-height: 1;
  top: 0;
  left: 0;
  font-size: 180px;
  color: #446DF644;
  -webkit-transform: translate(-100%,-50%);
  -ms-transform: translate(-100%,-50%);
  transform: translate(-100%,-50%);
  z-index: 0;
}

.c12 svg {
  position: absolute;
  width: 70px;
  fill: #446DF6;
  left: 0;
  top: 50%;
  -webkit-transform: translate(-50%,-30%);
  -ms-transform: translate(-50%,-30%);
  transform: translate(-50%,-30%);
}

.c16 {
  width: 70%;
  margin-bottom: 5rem;
}

.c13 {
  text-align: left;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-left: calc(15rem + 3%);
}

.c17 {
  padding: 1rem;
  margin-left: 5rem;
  margin-bottom: 5rem;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 5rem;
  background-color: #FFFFFF;
  border-radius: 5px;
  box-shadow: 0 0 10px -4px rgba(0,0,0,0.25);
  overflow: hidden;
}

.c5 {
  max-height: 10rem;
  margin-bottom: 3rem;
}

.c5 svg {
  width: 8rem !important;
  fill: #446DF6;
}

.c6 {
  font-size: 18px;
}

.c7 {
  position: absolute;
  bottom: -4%;
  line-height: 0.8;
  left: -2%;
  font-family: 'Montserrat';
  color: #446DF644;
  font-weight: 700;
  font-size: 180px;
}

.c3 {
  display: grid;
  grid-template-columns: repeat(3,1fr);
  grid-gap: 5%;
}

.c2 {
  width: 65vw;
}

.c2 > * {
  margin-bottom: 5rem;
}

.c0 {
  display: grid;
  grid-template-columns: 5rem [start] repeat(3,1fr) [end] 5rem;
  grid-auto-rows: minmax(min-content);
  grid-column-gap: 3%;
  margin: 0;
}

.c0 > *:not(section) {
  grid-column: start / end;
}

.c0 > section {
  grid-column: -1 / 1 !important;
}

.c23 {
  font-size: 64px;
  color: #F25C54;
  text-align: center;
  grid-column: start / end;
}

.c24 {
  margin: 0 auto;
  background-color: #FFFFFF;
  padding: 3rem 8.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: 700;
  border-radius: 5px;
  box-shadow: 0 0 10px -4px rgba(0,0,0,0.25);
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 20%;
  grid-column: 1 / -1 !important;
  height: 20rem;
  background: #446DF6;
  background-image: url(bg-pattern.png);
  background-position: bottom left;
}

.c18 > * {
  color: #FFFFFF !important;
}

@media (max-width:1300px) {
  .c22 {
    display: none;
  }
}

@media (max-width:680px) {
  .c8 {
    grid-template-columns: [start] 1fr [end];
    padding: 2rem;
  }
}

@media (max-width:680px) {
  .c14 {
    grid-template-columns: [start] 1fr [end];
    padding: 2rem;
  }
}

@media (max-width:1028px) {
  .c9 {
    grid-template-columns: 1fr;
    grid-column-gap: 0;
  }
}

@media (max-width:680px) {
  .c9 {
    position: relative;
    padding-top: 15rem;
  }
}

@media (max-width:680px) {
  .c11 {
    position: static;
  }
}

@media (max-width:680px) {
  .c15 {
    position: static;
  }
}

@media (max-width:680px) {
  .c12 {
    font-size: 180px;
    left: 50%;
    -webkit-transform: translate(-50%,0);
    -ms-transform: translate(-50%,0);
    transform: translate(-50%,0);
  }
}

@media (max-width:680px) {
  .c10 {
    padding-left: 0;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:1028px) {
  .c3 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (max-width:570px) {
  .c3 {
    grid-template-columns: 1fr;
    margin-bottom: 40rem;
  }
}

@media (max-width:680px) {
  .c2 {
    width: 100%;
    grid-column: 1 / -1;
    padding: 0 2rem;
  }
}

@media (max-width:680px) {
  .c23 {
    font-size: 34px;
  }
}

@media (max-width:680px) {
  .c24 {
    padding: 3rem 2rem;
  }
}

@media (max-width:680px) {
  .c18 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-evenly;
    -webkit-justify-content: space-evenly;
    -ms-flex-pack: space-evenly;
    justify-content: space-evenly;
    -webkit-align-items: space-evenly;
    -webkit-box-align: space-evenly;
    -ms-flex-align: space-evenly;
    align-items: space-evenly;
    padding: 0 2rem;
    text-align: center;
    height: 30rem;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      <h2>
        6 reasons to choose OMS invoicer
      </h2>
      <p>
        OMS is here for you! Adding customer, orders and invoices never been easier! Typos and simple mistakes on your invoices will never trouble you again! All customer data can be fetched from VIES database!
      </p>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg>
            clients-icon.svg
          </svg>
        </div>
        <h3
          class="c6"
        >
          Customers details in one place
        </h3>
        <div
          class="c7"
        >
          1
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg>
            orders-icon.svg
          </svg>
        </div>
        <h3
          class="c6"
        >
          Take control over your orders
        </h3>
        <div
          class="c7"
        >
          2
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg>
            invoice-icon.svg
          </svg>
        </div>
        <h3
          class="c6"
        >
          Easy invoicing
        </h3>
        <div
          class="c7"
        >
          3
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg>
            ok-icon.svg
          </svg>
        </div>
        <h3
          class="c6"
        >
          Instant VIES verification
        </h3>
        <div
          class="c7"
        >
          4
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg>
            cash-icon.svg
          </svg>
        </div>
        <h3
          class="c6"
        >
          Keep your payments under control
        </h3>
        <div
          class="c7"
        >
          5
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg>
            dashboard-icon.svg
          </svg>
        </div>
        <h3
          class="c6"
        >
          Useful dashboard notifications
        </h3>
        <div
          class="c7"
        >
          6
        </div>
      </div>
    </div>
  </section>
  <section>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <h3
            class="c11"
          >
            <div
              class="c12"
            >
              <svg>
                clients-icon.svg
              </svg>
              1
            </div>
            Customers details in one place
          </h3>
          
        </div>
        <div
          class="c13"
        >
          <p>
            Keep your customers details in one place. Address, VAT number, e-mail address and many more can be stored directly in customer cards!
            <br />
            <br />
            No matter how many customers your database will contain.
            <br />
            <b>
               Quick search 
            </b>
            in customer module gives you possibility to find any of them within a few seconds!
          </p>
        </div>
      </div>
    </div>
    <div
      class="c14"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <h3
            class="c11"
          >
            <div
              class="c12"
            >
              <svg>
                orders-icon.svg
              </svg>
              2
            </div>
            Take control over your orders
          </h3>
          
        </div>
        <div
          class="c13"
        >
          <p>
            Keep all of your
            <b>
               orders under control! 
            </b>
            <br />
            <br />
            You can now create orders for customers added to your database. Statuses will help with tracking proggres on each individual order.
          </p>
        </div>
      </div>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <h3
            class="c11"
          >
            <div
              class="c12"
            >
              <svg>
                invoice-icon.svg
              </svg>
              3
            </div>
            Easy invoicing
          </h3>
          
        </div>
        <div
          class="c13"
        >
          <p>
            Invoicing has
            <b>
               never been that easy!
            </b>
            <br />
            <br />
            Issue new invoice with just few mouse clicks. OMS invoicer will take all of neccesary data from orders and provide them on your bill! One thing that user has to do by himself is just to accept invoice draft!
          </p>
        </div>
      </div>
    </div>
    <div
      class="c14"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <h3
            class="c15"
          >
            <div
              class="c12"
            >
              <svg>
                ok-icon.svg
              </svg>
              4
            </div>
            Instant VIES verification
          </h3>
          <img
            alt="position image"
            class="c16"
            src="homepage-body-1.jpg"
          />
        </div>
        <div
          class="c13"
        >
          <p>
            Adding new customer by head is just a relic of past. With OMS you can fetch most important details about your customers directly from VIES database. To do that you have to just provide country prefix and new customer VAT EU number into invoices modal.
            <br />
            <b>
              What benefints do you have using VIES autofil?
            </b>
          </p>
          <ol
            class="c17"
          >
            <li>
              Save your time!
            </li>
            <li>
              Avoid mistakes!
            </li>
            <li>
              You will be always sure if your new customer have active VAT number
            </li>
            <li>
              It is just easier to do!
            </li>
          </ol>
          <span>
            Do not wait! Try it on your own!
          </span>
        </div>
      </div>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <h3
            class="c11"
          >
            <div
              class="c12"
            >
              <svg>
                cash-icon.svg
              </svg>
              5
            </div>
            Keep your payments under control
          </h3>
          
        </div>
        <div
          class="c13"
        >
          <p>
            Who did not have a problem with
            <b>
               untracked delayed payments? 
            </b>
            <br />
            <br />
            Thanks to invoice module you can mark which invoices are paid! By using search field you can also quickly filter all unpaid invoices to start collect your liabilities!
          </p>
        </div>
      </div>
    </div>
    <div
      class="c14"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <h3
            class="c15"
          >
            <div
              class="c12"
            >
              <svg>
                dashboard-icon.svg
              </svg>
              6
            </div>
            Useful dashboard notifications
          </h3>
          <img
            alt="position image"
            class="c16"
            src="homepage-body-2.jpg"
          />
        </div>
        <div
          class="c13"
        >
          <p>
            To wrap everyting in one place our dashbord will display most important informations. For example
          </p>
          <ol
            class="c17"
          >
            <li>
              Undpaid invoices
            </li>
            <li>
              Finished orders without invoice
            </li>
            <li>
              Pending orders
            </li>
            <li>
              Quick app nav
            </li>
          </ol>
          <p>
            .. and much more!
          </p>
        </div>
      </div>
    </div>
  </section>
  <div
    class="c18"
  >
    <a
      class="c19"
      href="/"
      linktype=""
    >
      <span
        class="c20"
      >
        <svg
          class="c21"
        >
          logo.svg
        </svg>
        <span
          class="c22"
        >
          Invoicer.v1
        </span>
      </span>
    </a>
    <h3>
      Are you ready to get started?
    </h3>
    <a
      class="c19"
      href="/register"
      linktype=""
    >
      Register now!
    </a>
  </div>
  <section
    class="c1"
  >
    <h2
      class="c23"
    >
      IMPORTANT!
    </h2>
    <div
      class="c24"
    >
      This app was created for learning purpuses. Please do not use it to issue real inovice or to keep real customer data!
    </div>
  </section>
</div>
`;
